### PR TITLE
rebar3 / Windows support / GH Action CI

### PR DESCRIPTION
**Pull request by kocolosk**
I wasn't sure if GH Actions could be developed in branches, but it sort of seems like they cannot. Here's a simple attempt to run unit tests on Pull Requests and every push to `main`.

### CODE REVIEW

**Comment by wohali** [comment]
@kocolosk FYI, doing it with GHA instead of Jenkins will put you in the global build queue for GHA for the ASF, which is rate limited across the board:

Average queue depth per repo @ ASF for 2021: https://cwiki.apache.org/confluence/download/attachments/191336468/image2021-11-1_14-33-54.png?version=1&modificationDate=1635773634000&api=v2

Number of ASF repos using GHA for 2021: https://cwiki.apache.org/confluence/download/attachments/191336468/image2021-11-1_14-28-47.png?version=1&modificationDate=1635773328000&api=v2

You are probably OK for now, but if the queue depths become a problem.... I know Jenkins stinks to use, but it's at least our dedicated hardware, and the pre-built Docker containers already support everything in `main`.

**Comment by kocolosk** [comment]
@wohali Yeah I saw those queue depths and was a bit surprised since my experience from the last few days has been that Actions are starting immediately. But clearly it's been an issue at times.

I'm totally fine using Jenkins, but I started down this GHA experiment because of Windows support. Do we have any view to adding a Windows executor to the CouchDB Jenkins setup?

File: .github/workflows/rebar.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          submodules: recursive
+      - name: Setup Erlang
+        uses: ./.github/actions/setup-beam
+        with:
+          otp-version: '24'
+          rebar3-version: '3.16'
```
Comment (vkatsuba) [inline .github/workflows/rebar.yml:65]: Can be used rebar version `3.17.0`.

File: .github/workflows/rebar.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    env:
+      DEBUG: 1
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          submodules: recursive
+      - name: Setup Erlang
+        uses: ./.github/actions/setup-beam
```
Comment (vkatsuba) [inline .github/workflows/rebar.yml:62]: I suppose can used just `erlef/setup-beam@v1` instead of submodule.

File: .github/workflows/rebar.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        with:
+          otp-version: '24'
+          rebar3-version: '3.16'
+      - name: Setup MSVC toolchain
+        uses: ./.github/actions/msvc-dev-cmd
```
Comment (vkatsuba) [inline .github/workflows/rebar.yml:67]:
```suggestion
        uses: ilammy/msvc-dev-cmd@v1
```

File: .github/workflows/rebar.yml (hunk 2 above)
Comment (kocolosk) [reply .github/workflows/rebar.yml:62]: Ah, I should add a comment to explain the approach here. The ASF actually blocked all direct usage of third-party actions in response to a security vulnerability late last year. The submodule pattern is the approved workaround that the Apache Airflow team designed. More details here: https://cwiki.apache.org/confluence/display/BUILDS/GitHub+Actions+status#GitHubActionsstatus-Security

File: .github/workflows/rebar.yml (hunk 3 above)
Comment (kocolosk) [reply .github/workflows/rebar.yml:67]: As above, Apache projects are blocked from using third-party actions directly.

File: .github/workflows/rebar.yml (hunk 2 above)
Comment (vkatsuba) [reply .github/workflows/rebar.yml:62]: Got it, thanks.

File: .github/workflows/rebar.yml (hunk 1 above)
Comment (kocolosk) [reply .github/workflows/rebar.yml]: Updated

File: .github/workflows/rebar.yml (hunk 2 above)
Comment (kocolosk) [reply .github/workflows/rebar.yml]: Added explanation in a2a5634ce58ccb8fff392e1a659c6403f3c390a5

**Comment by wohali** [comment]
not going to have time to look at this for a while - sorry about that. Don't let my lack of a review block this PR, and thanks for picking up the work on this Adam.

File: .gitignore [hunk 4]
```diff
@@ -1,12 +1,12 @@
-
+# Unit testing
+.erlfdb/
 .eunit/
-.rebar/
```
Comment (nickva) [inline .gitignore:3]: Minor not: running this as part of CouchDB main I still see `.rebar` and `ebin/` folders show up. Perhaps a side-effect of the main `repo` still using `rebar2`. We can leave a note perhaps to clean up `.gitignore` after we switch to `rebar3` in main.

**Review by nickva — APPROVED** [review]
I don't have easy access to a Windows dev environment but seeing as the GH actions CI passed and that main repo eunit and elixir tests pass I am +1 to merge it

File: .gitignore (hunk 4 above)
Comment (kocolosk) [reply .gitignore:3]: Good point. Addressed in 36e252b

**Pull request merged**